sound/soc/sof/ipc.c

```diff
@@ -623,39 +665,14 @@ int snd_sof_ipc_set_get_comp_data(struct snd_sof_ipc *ipc,
 {
 	struct snd_sof_dev *sdev = ipc->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
-	u32 send_bytes;
 	int send;
 	int err = 0;
 
 	/* read or write firmware volume */
 	if (scontrol->readback_offset != 0) {
 		/* write/read value header via mmaped region */
-		switch (ctrl_type) {
-		case SOF_CTRL_TYPE_VALUE_CHAN_GET:
-		case SOF_CTRL_TYPE_VALUE_COMP_GET:
-		case SOF_CTRL_TYPE_DATA_GET:
-			send = 0;
-			break;
-		case SOF_CTRL_TYPE_VALUE_CHAN_SET:
-		case SOF_CTRL_TYPE_VALUE_COMP_SET:
-		case SOF_CTRL_TYPE_DATA_SET:
-			send = 1;
-			break;
-		default:
-			return -EINVAL;
-		}
-		send_bytes = sizeof(struct sof_ipc_ctrl_value_chan) *
-		cdata->num_elems;
-
-		if (send)
-			snd_sof_dsp_block_write(sdev, sdev->mmio_bar,
-						scontrol->readback_offset,
-						cdata->chanv, send_bytes);
-
-		else
-			snd_sof_dsp_block_read(sdev, sdev->mmio_bar,
-					       scontrol->readback_offset,
-					       cdata->chanv, send_bytes);
+		send = sof_get_ctrl_ipc_dir(ctrl_type);
```

Comment:
need to check if send is < 0 since you can return -EINVAL

Reply:
ok done.